packages/forms-web-app/src/controllers/rule-6/enter-code.js

```diff
@@ -0,0 +1,197 @@
+const { createAppealUserSession } = require('../../services/user.service');
+const { isTokenValid } = require('../../lib/is-token-valid');
+const { enterCodeConfig } = require('@pins/common');
+const logger = require('../../lib/logger');
+
+const { getSessionEmail } = require('#lib/session-helper');
+const getAuthClient = require('@pins/common/src/client/auth-client');
+const config = require('../../config');
+const { AUTH } = require('@pins/common/src/constants');
+
+/**
+ * @typedef {import('#lib/is-token-valid').TokenValidResult} TokenValidResult
+ */
+
+/**
+ * @typedef {Object} enterCodeOptions
+ * @property {boolean} isGeneralLogin - defines if this enter code journey is for a general appeal log in, unrelated to an appeal
+ */
+
+/**
+ * Creates a one time password grant via the auth server
+ * @param {string} email
+ * @param {string} action
+ * @returns {Promise<void>}
+ */
+const createOTPGrant = async (email, action) => {
```

Comment:
lets reuse this function in auth-client perhaps

Reply:
Good idea - there is a fair amount of common code which could be refactored here I think.